src/isomorphic/classic/element/ReactCurrentRender.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright 2013-2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactCurrentRender
+ */
+
+'use strict';
+
+/**
+ * Keeps track of the currently rendering component. Not always the same as
+ * the current owner, which is also set on component initialization. Should
+ * only be updated in development, since it's only used for warnings.
+ */
+var ReactCurrentRender = {
```

Comment:
We don't need another thing to hold the identity of a component. We explicitly want to get rid of it completely. It would be better if this was just true/false if something is currently rendering since that should be all we need for warnings. This is important step forward because this file can live in the renderer package.

ReactCurrentOwner needs to live in some shared global package with a fixed version number so that it can be shared with ReactElement.

For now, we can use owner for the error message itself but we shouldn't track the instance here.

I understand that you need the instance to disambiguate.